notebooks/en/structured_generation.ipynb

```diff
@@ -0,0 +1,600 @@
+{
```

Comment:
<p>"This notebook first shows a naive approach to structured generation via prompting and <em>highlights its limits</em>, then demonstrates constrained decoding for more efficient structured generation"</p><p> </p><p>"It leverages Hugging Face Inference Endpoints (the example shows a <a href="https://huggingface.co/docs/api-inference/quicktour" target="_blank">serverless</a> endpoint, but you can directly change the endpoint to a <a href="https://huggingface.co/docs/inference-endpoints/en/guides/access" target="_blank">dedicated</a> one), then also <em>shows</em> a local pipeline using <a href="https://github.com/outlines-dev/outlines" target="_blank">outlines</a>, a structured text generation library."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/64/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>